### PR TITLE
Cleanup IO#popen

### PR DESCRIPTION
**Pull request by eregon**

### CODE REVIEW

File: src/main/ruby/core/io.rb [hunk 1]
```diff
@@ -880,7 +880,7 @@ def self.popen(*args)
     end
 
     mode, binary, external, internal, _autoclose =
-      IO.normalize_options(mode, io_options || {})
+      IO.normalize_options(mode, io_options)
```
Comment (bjfish) [inline src/main/ruby/core/io.rb]: Style question: Is it odd to end a line with `autoclose =` ?

File: src/main/ruby/core/io.rb [hunk 2]
```diff
@@ -921,31 +919,8 @@ def self.popen(*args)
...
-            exit! 0
-          else
-            return nil
-          end
-        rescue
-          exit! 0
-        end
-      end
+      Kernel.fork # will throw an error
+      raise 'unreachable'
```
Comment (bjfish) [inline src/main/ruby/core/io.rb]: Is `raise 'unreachable'` needed here?

File: src/main/ruby/core/io.rb (hunk 1 above)
Comment (eregon) [reply src/main/ruby/core/io.rb]: Yes, I meant to move the call to the same line, will do.

File: src/main/ruby/core/io.rb (hunk 2 above)
Comment (eregon) [reply src/main/ruby/core/io.rb]: Just safety in case somebody redefines `fork`.

**Pull request merged**